File: notifier-server/models/note.go
package models

import (
	"time"
)

type Note struct {
    NoteID    int       `json:"note_id"`
    UserID    int       `json:"user_id"`
    Title     string    `json:"title"`
    Content   string    `json:"content"`
    Deleted   bool      `json:"deleted"`
    CreatedAt time.Time `json:"created_at"`
    UpdatedAt *time.Time `json:"updated_at"`
}

File: notifier-client/src/utils/models/Note.ts
export interface Note {
  noteId: number | null
  userId: number
  title: string
  content: string
  deleted: boolean
  dirty: boolean
  createdAt: Date | null
  updatedAt: Date | null
}

export interface CreateNoteResult {
  success: boolean
  data?: Note
  error?: string
}

export interface GetNotesResult {
  success: boolean
  data?: Note[]
  error?: string
}

export interface UpdateNoteResult {
  success: boolean
  data?: Note
  error?: string
}

export interface NoteApiResponse {
  note_id: number | null
  user_id: number
  title: string
  content: string
  deleted: boolean
  created_at: Date | null
  updated_at: Date | null
}

export const transformNoteFromApi = (data: NoteApiResponse): Note => ({
  noteId: data.note_id,
  userId: data.user_id,
  title: data.title,
  content: data.content,
  deleted: data.deleted,
  createdAt: data.created_at,
  updatedAt: data.updated_at,
  dirty: false,
})
